### PR TITLE
Add unit tests for PinListSync

### PR DESCRIPTION
**Pull request by YangchenYe323**
### What changes are proposed in this pull request?
I added unit tests for `PinListSync`

### Why are the changes needed?
It improves unit-test coverage of the class (0% -> 100%)

### Does this PR introduce any user facing changes?
No

### CODE REVIEW

File: core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    mSync = mCloser.register(new PinListSync(mBlockWorker, mFileSystemMasterClient));
+  }
+
+  @Test
+  public void syncPinList() throws Exception {
+    Set<Long> testPinLists = ImmutableSet.of(1L, 2L, 3L);
+    doReturn(testPinLists)
+        .when(mFileSystemMasterClient)
+        .getPinList();
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java:50]: Can you replace mock with a subclass of `FileSystemMasterClient` which overrides the impl of `getPinList`. Should not be too hard.

File: core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    // should receive the latest pin list
+    assertEquals(testPinLists, mBlockWorker.getPinList());
+  }
+
+  @Test
+  public void syncPinListFailure() throws Exception {
+    // simulate failure retrieving pin list from master
+    doThrow(new IOException())
+        .when(mFileSystemMasterClient)
+        .getPinList();
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java:63]: same as above

File: core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    // should fail
+    mSync.heartbeat();
+
+    // should not get any pin list update
+    assertEquals(ImmutableSet.of(), mBlockWorker.getPinList());
+  }
+
+  @After
+  public void after() throws Exception {
+    mCloser.close();
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java:74]: No need to use a closer here. `mSync.close()` should do.

**Review by dbw9580 — CHANGES_REQUESTED** [review]

File: core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java (hunk 3 above)
Comment (YangchenYe323) [reply core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java]: Done

File: core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java [hunk 4]
```diff
@@ -29,26 +28,33 @@
...
-  private final Closer mCloser = Closer.create();
 
   @Before
   public void before() {
     mBlockWorker = new TestBlockWorker();
-    mFileSystemMasterClient = mock(FileSystemMasterClient.class);
-
-    mSync = mCloser.register(new PinListSync(mBlockWorker, mFileSystemMasterClient));
+    mFileSystemMasterClient = null;
+    mSync = null;
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java:43]: In this case these do not have to be class fields.

File: core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java [hunk 5]
```diff
@@ -29,26 +28,33 @@
...
+      Configuration.global())).build();
+
   private PinListSync mSync;
   private TestBlockWorker mBlockWorker;
   private FileSystemMasterClient mFileSystemMasterClient;
-  private final Closer mCloser = Closer.create();
 
   @Before
   public void before() {
     mBlockWorker = new TestBlockWorker();
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java:41]: Just initialize this at the place where this is declared.

Comment (YangchenYe323) [reply core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java:38]: Fixed

File: core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java (hunk 4 above)
Comment (YangchenYe323) [reply core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java]: Done

File: core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+  // context for creating FileSystemMasterClient
+  private final MasterClientContext mContext = MasterClientContext.newBuilder(ClientContext.create(
+      Configuration.global())).build();
+
+  private TestBlockWorker mBlockWorker;
+
+  @Before
+  public void before() {
+    mBlockWorker = new TestBlockWorker();
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java:38]: This is redundant. Initialize where it is declared.

**Review by dbw9580 — APPROVED** [review]
LGTM except for a minor issue.

File: core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java (hunk 6 above)
Comment (dbw9580) [reply core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java:38]: I believe JUnit creates a new instance of the test class for every `@Test`. So initializing things in constructor is the same as in a `@Before` method. [reference](https://stackoverflow.com/q/6094081)

**Comment by YangchenYe323** [comment]
@jiacheliu3 Do you mind taking a look at this? Thanks!

File: core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java (hunk 6 above)
Comment (YangchenYe323) [reply core/server/worker/src/test/java/alluxio/worker/block/PinListSyncTest.java:38]: Yeah that is actually the case. I wonder why they would like doing this though...

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**